objax/functional/loss.py

```diff
@@ -38,6 +39,9 @@ def cross_entropy_logits(logits: JaxArray, labels: JaxArray) -> JaxArray:
     Returns:
         (batch, ...) tensor of the cross-entropies for each entry.
     """
+    if logits.shape != labels.shape:
+        warnings.warn('Size Mismatch: logits {} and labels {}'.format(logits.shape, labels.shape))
```

Comment:
Could you please add function name into each warning?
Like following: `'Argument shape mismatch in cross_entropy_logits: logits {} and labels {}'.format(logits.shape, labels.shape)'`

Reply:
Hey @AlexeyKurakin have made the following changes as per your comment. I added a separate decorator for this purpose in order to keep the code simple. Kindly review the changes and merge.

Reply:
I would actually ask to use simple warnings instead of decorator. Generally decorators in Python adds a level of indirection, thus understanding the code will require looking at the code of the decorator.
In Objax we prefer to have explicit code whenever possible. This makes it easier to know what code is doing and easier to debug code.  As far I as remember, the only place where we use and encourage decorator is `objax.Function.with_vars` or `objax.Function.auto_vars`.

Also I would ask to avoid overriding warnings formatter. User of the library may have their own customization of warning formatter, which will interfere with overrides done in Objax.

Reply:
Ohh then would change it to the old way. No issues.